entities/mimic/mimic_class.gd

```diff
@@ -92,9 +98,6 @@ func attack() -> void:
 	set_state(State.NONE)
 	set_state(State.ATTACK)
 
-func collect():
```

Comment:
I added this empty `collect()` to avoid a crash if the hero starts collecting the Mimic in chest form, and the Mimic transforms back before the hero is done (therefore it's no longer collectible and fails the `assert()` in Entity.collect())